### PR TITLE
added support for property-based ANSI coloring

### PR DESCRIPTION
**Pull request by sailorgeoffrey**
Slightly cleaner implementation of my previously submitted request.  This uses the existing AnsiElement already in boot

### CODE REVIEW

File: spring-boot/src/main/java/org/springframework/boot/ResourceBanner.java [hunk 1]
```diff
@@ -77,7 +87,8 @@ public void printBanner(Environment environment, Class<?> sourceClass, PrintStre
 		List<PropertyResolver> resolvers = new ArrayList<PropertyResolver>();
 		resolvers.add(environment);
 		resolvers.add(getVersionResolver(sourceClass));
-		return resolvers;
+        resolvers.add(getAnsiResolver());
```
Comment (spencergibb) [inline spring-boot/src/main/java/org/springframework/boot/ResourceBanner.java]: boot uses tabs

Comment (sailorgeoffrey) [inline spring-boot/src/main/java/org/springframework/boot/ResourceBanner.java]: fixed

File: spring-boot/src/test/java/org/springframework/boot/ResourceBannerTests.java [hunk 2]
```diff
@@ -29,6 +29,7 @@
 import org.springframework.mock.env.MockEnvironment;
 
 import static org.hamcrest.Matchers.startsWith;
+import static org.junit.Assert.assertEquals;
```
Comment (spencergibb) [inline spring-boot/src/test/java/org/springframework/boot/ResourceBannerTests.java]: unused?

**Comment by philwebb** [comment]
Perhaps I asked already, but can you confirm that you have signed the CLA.

**Comment by sailorgeoffrey** [comment]
Yes I have.

**Comment by sailorgeoffrey** [comment]
I noticed other ANSI work in the project and thought it would be best to leverage that.  It is now strongly typed to an enum that is shared with the other items in spring boot.

**Comment by sailorgeoffrey** [comment]
I've got everything in there that I want... Not sure what the next step is.

**Comment by philwebb** [comment]
We plan to merge this for 1.3. We'll let you know if anything needs changing when we merge the code. Thanks for the contribution.

**Comment by philwebb** [comment]
Thanks for the pull-request. I made a few changes before merging it:

* I like the idea of making `AnsiElement` an enum but it could break existing application. Instead I've deprecated the constants and introduced `AnsiStyle`, `AnsiColor` and `AnsiBackground` enums.
* I've moved the `PropertyResolver` logic from `ResourceBanner` to the `ansi` package, that also means we don't need to make so of `AnsiElement` public.
* The property names have changed a little, you can now use `${Ansi.XXX}`, `${AnsiStyle.XXX}`, `${AnsiColor.XXX}` or `${AnsiBackground.XXX}`.

Cheers!

**Comment by johntostring** [comment]
I used `${AnsiColor.RED}` in my banner.txt
Now the console print log all red.

**Comment by snicoll** [comment]
@johntostring this issue is closed. If you have a question, please raise a StackOverflow thread. Thanks!

**Comment by wilkinsona** [comment]
@johntostring To add to what @snicoll said. I suspect you haven't used `${AnsiElement.NORMAL}` to switch back to the normal colour.

**Pull request closed without merging**